src/abi.js

```diff
@@ -34,17 +34,31 @@ module.exports = class ABI {
   }
 
   static oldEventSignature(event) {
-    return `${event.get('name')}(${(event.get('inputs') || [])
+    return `${event.get('name')}(${event
+      .get('inputs', [])
       .map(input => buildOldSignatureParameter(input))
       .join(',')})`
   }
 
   static eventSignature(event) {
-    return `${event.get('name')}(${(event.get('inputs') || [])
+    return `${event.get('name')}(${event
+      .get('inputs', [])
       .map(input => buildSignatureParameter(input))
       .join(',')})`
   }
 
+  functionSignature(fn) {
+    let inputs = fn
+      .get('inputs', [])
+      .map(buildSignatureParameter)
+      .join(',')
+    let outputs = fn
+      .get('outputs', [])
+      .map(buildSignatureParameter)
+      .join(',')
+    return `${fn.get('name')}(${inputs})${outputs.length > 0 ? `:(${outputs})` : ''}`
```

Comment:
It could be helpful for readability to include a comment explaining the function signature format since the `return ...` line is a bit hard to parse.

Reply:
I've added the following comment above the method:
```js
  /**
   * For the ABI of a function, returns a string function signature compatible
   * with the Rust `ethabi` library. It is of the form
   *
   *     <function>([<input-type-1>, ...])[:(<output-type-1,...)]
   *
   * A few examples for a function called `example`:
   *
   * - No inputs or outputs: `example()`
   * - One input and output: `example(uint256):(bool)`
   * - Multiple inputs and outputs: `example(uint256,(string,bytes32)):(bool,uint256)`
   */
```

Reply:
Perfect